workflows/fastquorum.nf

```diff
@@ -1,211 +1,209 @@
+    FASTQC(
+        ch_samplesheet
     )
     ch_versions = ch_versions.mix(FASTQC.out.versions.first())
-
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
-    )
+    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]})
 
     //
     // MODULE: Run fgbio FastqToBam
     //
-    FASTQTOBAM(INPUT_CHECK.out.reads)
+    FASTQTOBAM(ch_samplesheet)
+    ch_versions = ch_versions.mix(FASTQTOBAM.out.versions.first())
 
     //
     // MODULE: Align with bwa mem
     //
-    grouped_sort = true
-    ALIGN_RAW_BAM(FASTQTOBAM.out.bam, ch_ref_index_dir, grouped_sort)
+    ALIGN_RAW_BAM(FASTQTOBAM.out.bam, ch_fasta, ch_fasta_fai, ch_dict, ch_bwa, "template-coordinate")
+    ch_versions = ch_versions.mix(ALIGN_RAW_BAM.out.versions.first())
+
+    //
+    // Create a channel that:
+    // 1. Groups the aligned BAMs by sample identifier.  We use `groupKey` here since we know how many BAMs each
+    //    sample expects to have.  Typically a sample has more than one BAM if it had multiple runs or lanes.
+    // 2. Splits the samples into those that have more than one BAM, and those that have exactly one BAM.  The former
+    //    samples will have their BAMs merged.
+    //
+    bam_to_merge = ALIGN_RAW_BAM.out.bam
+        .map {
+            meta, bam ->
+                [ groupKey(meta, meta.n_samples), bam ]
+    }
+    .groupTuple()
+    .branch { meta, bam ->
+        // The `n_samples` is added by `validateInputSamplesheet` method in `PIPELINE_INITIALISATION` workflow
+        single:   meta.n_samples <= 1
+            return [ meta, bam[0] ]  // NB: bam is a list (of one BAM) so return just the one BAM
+        multiple: meta.n_samples > 1
+    }
+
+    //
+    // MODULE: Run samtools merge to merge across runs/lanes for the same sample
+    //
+    MERGE_BAM(bam_to_merge.multiple, [[], []], [[], []])
+    ch_versions = ch_versions.mix(MERGE_BAM.out.versions.first())
+
+    //
+    // Create a channel that contains the merged BAMs and those that did not need to be merged.
+    //
+    bam_all = MERGE_BAM.out.bam.mix(bam_to_merge.single)
 
     //
     // MODULE: Run fgbio GroupReadsByUmi
     //
-    GROUPREADSBYUMI(ALIGN_RAW_BAM.out.bam, groupreadsbyumi_strategy, params.groupreadsbyumi_edits)
+    GROUPREADSBYUMI(bam_all, params.groupreadsbyumi_strategy, params.groupreadsbyumi_edits)
+    ch_multiqc_files = ch_multiqc_files.mix(GROUPREADSBYUMI.out.histogram.map{it[1]}.collect())
+    ch_versions = ch_versions.mix(GROUPREADSBYUMI.out.versions.first())
 
-    // TODO: duplex_seq can be inferred from the read structure, but that's out of scope for now
     if (params.duplex_seq) {
         //
-        // MODULE: Run fgbio CallDuplexConsensusReads
+        // MODULE: Run fgbio CollecDuplexSeqMetrics
         //
-        CALLDDUPLEXCONSENSUSREADS(GROUPREADSBYUMI.out.bam, call_min_reads, params.call_min_baseq)
+        COLLECTDUPLEXSEQMETRICS(GROUPREADSBYUMI.out.bam)
```

Comment:
Are there any metrics from here that could go into multiqc?

Reply:
https://github.com/nf-core/fastquorum/issues/46